### ISSUE TITLE
CNN input question!

### ISSUE THREAD

**Issue by gcmy** [inspirehep/magpie#124]
Hi,I  want to know why there are 5 input layers in your CNN models,
current_input = Input(shape=(SAMPLE_LENGTH, embedding_size)),currently on the web, I just find the length of input layer is 1.
thanks!

**Comment by jstypka** [comment]
Hey @gcmy , it's because I use the network model described in [this paper](https://arxiv.org/abs/1408.5882) that uses 5 different ngram window sizes. Feel free to take a look at the paper if you want to understand it better - there is a nice picture about it!

**Comment by gcmy** [comment]
Many thanks,I have just read the paper about textCNN,But now I have a question about part of metrix(I think this is the target function we needed),in your project ‘’top_k_categorical_accracy‘’ were used,but it means that as least one target category exists in the first k values of the predicted value, the prediction is correct.Actually ,we need to know how many labels were correct in the essay with more than one labels,so the metrix should be this:
![the score formula](https://user-images.githubusercontent.com/20899440/34601548-6a693f70-f237-11e7-9b7a-8b19b4fc7157.JPG)
So I alter the function like this: 
```
'''
Top_k_categorical_accuracy is the number of labels in one essay
return:scores
'''
   top_K = tf.nn.top_k(y_true,Top_k_categorical_accuracy)
    top_Y = 0
    for top_i in range(Top_k_categorical_accuracy):
        top_Y += K.mean(K.in_top_k(y_pred,top_K[1][:,top_i],k=Top_k_categorical_accuracy))
    return top_Y/Top_k_categorical_accuracy
```
But it exists a problem.It only applied to the situation with the same number of labels .Such as all the essay have 3 labels.
So I want to ask you how to deal with this.with many thanks!

**Comment by jstypka** [comment]
@gcmy I'm not sure I understand your question very well, but it seems to me like you're looking for a [recall](https://en.wikipedia.org/wiki/Precision_and_recall) metric.

This metric computes how many of all the correct labels for a document did you predict as one. It can also be reduced to only the top K i.e. recall of 0.33 means that only a third of all true labels (`y_true`) were ranked in the top K documents. Let me know if that makes sense!

**Comment by gcmy** [comment]
@jstypka yep,but there is a little difference,The K values is not stable,it should be the true number of labels, for example,one document has 4 labels(A1,B1,C1,D1),the predicted labels is (B1,C1,F1,G1),and another document is (A2,B2,C2), the predicted labels is (B2,E2,F2),then the recall metric is (2+1)/(4+3)=3/7

**Comment by jstypka** [comment]
Ahhh, then I don't think it's a standard metric - you'd need to implement it yourself. I would advise copy pasting from [Keras recall implementation](https://github.com/keras-team/keras/blob/1.2.2/keras/metrics.py#L128-L131) and adjust it for your purposes. Good luck!

**Comment by gcmy** [comment]
@jstypka thank you,I decided to use the F_value to replace the metric.now i am waiting for the answer
